synapse/http/client.py

```diff
@@ -226,7 +227,9 @@ def resolutionComplete() -> None:
         return recv
 
 
-@implementer(ISynapseReactor)
+# ISynapseReactor implies IReactorCore, but explicitly marking it this as an implementer
+# of IReactorCore seems to keep mypy-zope happier.
+@implementer(IReactorCore, ISynapseReactor)
```

Comment:
It's strange that we have to do this, but oh well.

Reply:
Agreed.

For completeness, without this change (and the similar one for `ThreadedMemoryReactorClock` I get:

```
$ mypy
tests/http/test_proxyagent.py:689: error: Argument 1 to "ProxyAgent" has incompatible type "BlacklistingReactorWrapper"; expected "IReactorCore"  [arg-type]
tests/http/test_proxyagent.py:690: error: Argument 1 to "BlacklistingReactorWrapper" has incompatible type "ThreadedMemoryReactorClock"; expected "IReactorPluggableNameResolver"  [arg-type]
tests/http/test_proxyagent.py:735: error: Argument 1 to "ProxyAgent" has incompatible type "BlacklistingReactorWrapper"; expected "IReactorCore"  [arg-type]
tests/http/test_proxyagent.py:736: error: Argument 1 to "BlacklistingReactorWrapper" has incompatible type "ThreadedMemoryReactorClock"; expected "IReactorPluggableNameResolver"  [arg-type]
Found 4 errors in 1 file (checked 775 source files)
```

which is unfortunate.